test/e2e/data/infrastructure-docker/v1beta1/main/clusterclass-quick-start.yaml

```diff
@@ -268,29 +268,29 @@ spec:
           readOnly: true
           pathType: "File"
       - op: add
-        path: "/spec/template/spec/kubeadmConfigSpec/files"
+        path: "/spec/template/spec/kubeadmConfigSpec/files/-"
```

Comment:
Tests are failing with "add operation does not apply: doc is missing path: \"/spec/template/spec/kubeadmConfigSpec/files/-\" in the CAPI controller

I think you have to add an empty `files` array to the KubeadmControlPlaneTemplate (in both ClusterClasses)